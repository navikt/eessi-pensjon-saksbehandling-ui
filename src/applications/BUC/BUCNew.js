import React from 'react'

import { Store } from './index'
import * as constants from './constants'

import { Hovedknapp } from '../../components/ui/Nav'

const BUCNew = (props) => {

  const { state, dispatch } = React.useContext(Store)

  const onBUCList = () => {
    dispatch({
      type: constants.BUC_MODE,
      payload: 'list'
    })
  }

  return <div>
    <Hovedknapp onClick={onBUCList}>List</Hovedknapp>
  </div>
}

export default BUCNew
