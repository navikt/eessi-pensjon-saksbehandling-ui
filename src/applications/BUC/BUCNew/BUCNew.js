import React from 'react'

import { Store } from '../index'
import * as constants from '../constants'

import { Hovedknapp, Panel, Ingress } from '../../../components/ui/Nav'

const BUCNew = (props) => {

  const { state, dispatch } = React.useContext(Store)
  const { t } = props

  const onBUCList = () => {
    dispatch({
      type: constants.BUC_MODE_SET,
      payload: 'list'
    })
  }

  return <div>
    <div className='text-right'>
      <Hovedknapp onClick={onBUCList}>{t('buc:backToList')}</Hovedknapp>
    </div>
    <Panel>
      <Ingress>{t('buc:startNewBUC')}</Ingress>
      <hr/>

    </Panel>
  </div>
}

export default BUCNew
