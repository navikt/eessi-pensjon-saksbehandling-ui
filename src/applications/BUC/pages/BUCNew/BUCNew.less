@import '../../../../constants';

.a-buc-p-bucnew {
  padding: 2rem 5rem 2rem 5rem !important;

  .label {
    display:  flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-end;
  }
}


.highContrast .a-buc-p-bucnew {
  background: @black;
}
