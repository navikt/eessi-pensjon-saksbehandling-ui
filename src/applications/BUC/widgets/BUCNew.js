import React from 'react'
import { Hovedknapp, Panel, Systemtittel } from 'components/ui/Nav'
import BUCStart from 'applications/BUC/components/BUCStart/BUCStart'

const BUCNew = (props) => {
  const { t, actions } = props

  const onBUCList = () => {
    actions.resetBuc()
    actions.setMode('list')
  }

  return <React.Fragment>
    <div className='a-buc-buclist__buttons mt-3 mb-3'>
      <Hovedknapp onClick={onBUCList}>{t('ui:back')}</Hovedknapp>
    </div>
    <Panel>
      <Systemtittel>{t('buc:step-startBUCTitle')}</Systemtittel>
      <hr />
      <BUCStart mode='widget' {...props} />
    </Panel>
  </React.Fragment>
}

export default BUCNew
