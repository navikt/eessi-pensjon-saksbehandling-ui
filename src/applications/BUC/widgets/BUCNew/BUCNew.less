.a-buc-w-bucnew {
  border: 1px solid #59514B;
  padding: 2rem 5rem 2rem 5rem !important;

  .label {
    display:  flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: flex-end;
  }
}
