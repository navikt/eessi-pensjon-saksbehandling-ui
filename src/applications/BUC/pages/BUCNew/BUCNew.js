import React from 'react'
import PT from 'prop-types'
import { Nav } from 'eessi-pensjon-ui'
import BUCStart from 'applications/BUC/components/BUCStart/BUCStart'
import BUCFooter from 'applications/BUC/components/BUCFooter/BUCFooter'
import './BUCNew.css'

const BUCNew = (props) => {
  const { t } = props
  return (
    <Nav.Panel className='a-buc-bucnew s-border'>
      <Nav.Systemtittel>{t('buc:step-startBUCTitle')}</Nav.Systemtittel>
      <hr />
      <BUCStart mode='widget' {...props} />
      <BUCFooter className='w-100 mb-2' {...props} />
    </Nav.Panel>
  )
}

BUCNew.propTypes = {
  t: PT.func.isRequired
}
export default BUCNew
