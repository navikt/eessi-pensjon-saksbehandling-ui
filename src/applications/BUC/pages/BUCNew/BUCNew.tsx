import BUCFooter from 'applications/BUC/components/BUCFooter/BUCFooter'
import BUCStart, { BUCStartProps } from 'applications/BUC/components/BUCStart/BUCStart'
import Ui from 'eessi-pensjon-ui'
import PT from 'prop-types'
import React from 'react'
import './BUCNew.css'
import { useTranslation } from 'react-i18next'

export interface BUCNewProps extends BUCStartProps {
  aktoerId: string;
  setMode: (s: string) => void;
}

const BUCNew: React.FC<BUCNewProps> = ({ aktoerId, setMode } : BUCNewProps): JSX.Element => {
  const { t } = useTranslation()
  return (
    <>
      <Ui.Nav.Panel className='a-buc-p-bucnew s-border'>
        <Ui.Nav.Systemtittel>{t('buc:step-startBUCTitle')}</Ui.Nav.Systemtittel>
        <hr />
        <BUCStart aktoerId={aktoerId} setMode={setMode} />
      </Ui.Nav.Panel>
      <BUCFooter className='w-100 mt-2 mb-2' />
    </>
  )
}

BUCNew.propTypes = {
  aktoerId: PT.string.isRequired,
  setMode: PT.func.isRequired
}
export default BUCNew
