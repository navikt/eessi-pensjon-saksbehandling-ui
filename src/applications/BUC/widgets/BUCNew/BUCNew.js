import React from 'react'
import PT from 'prop-types'
import { Nav } from 'eessi-pensjon-ui'
import BUCStart from 'applications/BUC/components/BUCStart/BUCStart'

import './BUCNew.css'

const BUCNew = (props) => {
  const { t } = props
  return (
    <Nav.Panel className='a-buc-bucnew s-border'>
      <Nav.Systemtittel>{t('buc:step-startBUCTitle')}</Nav.Systemtittel>
      <hr />
      <BUCStart mode='widget' {...props} />
    </Nav.Panel>
  )
}

BUCNew.propTypes = {
  t: PT.func.isRequired
}
export default BUCNew
