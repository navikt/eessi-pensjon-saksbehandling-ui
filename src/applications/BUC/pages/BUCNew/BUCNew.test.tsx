import { mount, ReactWrapper } from 'enzyme'
import React from 'react'
import BUCNew, { BUCNewProps } from './BUCNew'

jest.mock('applications/BUC/components/BUCStart/BUCStart', () => {
  return (props: any) => (<div className='mock-bucstart' title={props.mode} />)
})

describe('applications/BUC/widgets/BUCNew/BUCNew', () => {
  let wrapper: ReactWrapper
  const initialMockProps: BUCNewProps = {
    actions: {},
    t: jest.fn(t => t),
    loading: {},
    locale: 'nb',
    setMode: jest.fn(),
    onTagsChanged: jest.fn()
  }

  it('Renders', () => {
    wrapper = mount(<BUCNew {...initialMockProps} />)
    expect(wrapper.isEmptyRender()).toBeFalsy()
    expect(wrapper).toMatchSnapshot()
  })

  it('Has proper HTML structure', () => {
    wrapper = mount(<BUCNew {...initialMockProps} />)
    expect(wrapper.exists('.a-buc-p-bucnew')).toBeTruthy()
    expect(wrapper.exists('.mock-bucstart')).toBeTruthy()
  })
})
