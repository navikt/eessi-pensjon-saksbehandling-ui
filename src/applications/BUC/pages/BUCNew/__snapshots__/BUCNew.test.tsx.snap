// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/widgets/BUCNew/BUCNew Renders 1`] = `
<BUCNew
  aktoerId="123"
  setMode={[MockFunction]}
>
  <Panel
    border={false}
    className="a-buc-p-bucnew s-border"
  >
    <div
      className="panel a-buc-p-bucnew s-border"
    >
      <Systemtittel>
        <TypografiBase
          type="systemtittel"
        >
          <h2
            className="typo-systemtittel"
          >
            buc:step-startBUCTitle
          </h2>
        </TypografiBase>
      </Systemtittel>
      <hr />
      <Component
        aktoerId="123"
        setMode={[MockFunction]}
      >
        <div
          className="mock-bucstart"
        />
      </Component>
    </div>
  </Panel>
  <Component
    className="w-100 mt-2 mb-2"
  >
    <div
      className="mock-bucfooter"
    />
  </Component>
</BUCNew>
`;
